tests/specifications/googlefonts_test.py

```diff
@@ -135,50 +135,118 @@ def test_example_checkrunner_based(cabin_regular_path):
       break
 
 
+class CheckResultsEvaluation():
+
+  def __init__(self, log_messages):
+    self.log_messages = log_messages
+
+  def assert_result(self, expected_status, expected_message=None):
+    last_result = None
+    for result in self.log_messages:
+      status, message, _ = result
+      if status in check_statuses:
+        last_result = result
+      if status == ENDCHECK:
+        assert last_result # The check must have emitted
+                           # at least one log message!
+
+        # we always test at least the status:
+        assert last_result[0] == expected_status
+
+        if expected_message: # and testing the message is optional
+          assert last_result[1] == expected_message
+        break
+
+
+class FBCheckTesterBadParamsError(Exception):
+  pass
+
+
+class FBCheck_Tester():
+
+  def __init__(self, profile, checkid):
+    self.profile = profile
+    self.checkid = checkid
+    self.log_messages = None
+    self.testcase_number = 0
+    print(f"Testing check '{checkid}' from the '{profile}' profile.")
+
+  def run(self, params, reasoning):
+    from fontbakery.checkrunner import CheckRunner
+    assert self.profile == "googlefonts" #TODO: use import module to load the correct specification otherwise.
```

Comment:
For tests, I think having an explicit import of the profile/spec may be better. It feels like the auto magic of hidden/implicit module loading defies a bit the purpose of documenting and the need for clarity of unit tests. Hence, how about doing sth. like this:

```py
from somewhere import FBCheckTester
from fontbakery.specifications.googlefonts import specification

GooglefontsFBCheckTester = FBCheckTester.partialConstructor('Googlefonts', specification)

# then in the test functions:

check = GooglefontsFBCheckTester("com.google.fonts/check/001")

# where FBCheckTester defines partialConstructor:

from functools import partialmethod
class FBCheckTester(object):
   def __init__(self, profile, checkid):
    # etc.
    pass

  @classmethod
  def partialConstructor(cls, name, *args, **kwds):
    class PartialCtor(cls):
      __init__ = partialmethod(cls.__init__, *args, **kwds)
    PartialCtor.__name__ = f'{name}{cls.__name__}'
    return NewCls

```